ibis/backends/tests/test_array.py

```diff
@@ -72,6 +73,75 @@
 # list.
 
 
+def test_array_factory(con):
+    a = ibis.array([1, 2, 3])
+    assert a.type() == dt.Array(value_type=dt.Int8)
+    assert con.execute(a) == [1, 2, 3]
+
+    a2 = ibis.array(a)
+    assert a.type() == dt.Array(value_type=dt.Int8)
+    assert con.execute(a2) == [1, 2, 3]
+
+
+def test_array_factory_typed(con):
+    typed = ibis.array([1, 2, 3], type="array<string>")
```

Comment:
I don't really like the implicit conversion here. Is it necessary?

Reply:
The conversion is exactly the part of the API I was trying to test here. I think this conversion is something that this API should explicitly support. What do you think?

Maybe I add a param as

```python
@pytest.mark.param(raw, [[1,2,3], ["1", "2", "3"]])
def test_array_factory_typed(con, raw):
    typed = ibis.array(raw, type="array<string>")
    ...
```